packages/autocomplete-js/src/__tests__/panelPlacement.test.ts

```diff
@@ -1,3 +1,233 @@
+
+      await waitFor(() => {
+        expect(document.querySelector('.aa-Panel')).toHaveStyle({
+          top: '9px', // OFFSET_TOP + HEIGHT
+          left: '11px', // LEFT
+        });
+      });
+    });
+  });
+
+  describe('with `end` value', () => {
+    test('places the panel bellow the input and align it with the input right side', async () => {
+      autocomplete({
+        container,
+        panelPlacement: 'end',
+        initialState: {
+          isOpen: true,
+        },
+      });
+
+      // Mock `getBoundingClientRect` for elements used in the panel placement calculation
+      document.querySelector(
+        '.aa-Form'
+      ).getBoundingClientRect = mockedGetBoundingClientRect;
+      document.querySelector(
+        '.aa-Autocomplete'
+      ).getBoundingClientRect = mockedGetBoundingClientRect;
+
+      await waitFor(() => {
+        expect(document.querySelector('.aa-Panel')).toHaveStyle({
+          top: '9px', // OFFSET_TOP + HEIGHT
+          right: '-27px', // CLIENT_WIDTH - (LEFT + WIDTH)
+        });
+      });
+    });
+  });
+
+  describe('with `full-width` value', () => {
+    test('places the panel bellow the input and takes the full page width', async () => {
+      autocomplete({
+        container,
+        panelPlacement: 'full-width',
+        initialState: {
+          isOpen: true,
+        },
+      });
+
+      // Mock `getBoundingClientRect` for elements used in the panel placement calculation
+      document.querySelector(
+        '.aa-Form'
+      ).getBoundingClientRect = mockedGetBoundingClientRect;
+      document.querySelector(
+        '.aa-Autocomplete'
+      ).getBoundingClientRect = mockedGetBoundingClientRect;
+
+      await waitFor(() => {
+        expect(document.querySelector('.aa-Panel')).toHaveStyle({
+          top: '9px', // OFFSET_TOP + HEIGHT
+          left: 0,
+          right: 0,
+          width: 'unset',
+          'max-width': 'unset',
+        });
+      });
+    });
+  });
+
+  test('behaves like `input-wrapper-width` if no value is set', async () => {
+    autocomplete({
+      container,
+      initialState: {
+        isOpen: true,
+      },
+    });
+
+    // Mock `getBoundingClientRect` for elements used in the panel placement calculation
+    document.querySelector(
+      '.aa-Form'
+    ).getBoundingClientRect = mockedGetBoundingClientRect;
+    document.querySelector(
+      '.aa-Autocomplete'
+    ).getBoundingClientRect = mockedGetBoundingClientRect;
+
+    await waitFor(() => {
+      expect(document.querySelector('.aa-Panel')).toHaveStyle({
+        top: '9px', // OFFSET_TOP + HEIGHT
+        left: '11px', // LEFT
+        right: '-27px', // CLIENT_WIDTH - (LEFT + WIDTH)
+        width: 'unset',
+        'max-width': 'unset',
+      });
+    });
+  });
+
+  // eslint-disable-next-line jest/no-done-callback
+  test('throws an error if an invalid value is used', async (done) => {
+    window.addEventListener(
```

Comment:
Why not use [`toThrow`](https://jestjs.io/docs/en/expect#tothrowerror) here?

Reply:
The error is throwed asyncronously so we cannot catch it with `toThrow`, we have to rely on the global error listener.